### PR TITLE
Feature/delete list

### PR DESCRIPTION
**Pull request by ender554**
Delete working

### CODE REVIEW

File: src/components/ShoppingList/ShoppingList.js [hunk 1]
```diff
@@ -1,28 +1,78 @@
...
+
+  confirmDelete() {
+    this.props.deleteList(this.props.id);
+    this.setState({
+      deleteModal: !this.state.deleteModal,
+    });
+  }
+
+  render() {
+    const { id, store, name } = this.props;
```
Comment (aseehra) [inline src/components/ShoppingList/ShoppingList.js]: So, I don't know why this doesn't error out in the edit-list-name branch, but I get a React-Redux error that passing the redux store is not allowed.

This is due to the fact that React-Redux treats the `store` prop as a reserved word. As result, you will have to use a different name (`groceryStore`?) for the prop. If you don't want to change your code too much, change this line to

```javascript
const { id, groceryStore: store, name } = this.props;
```

Just be sure to change the prop that is passed to this component in Lists.js.

File: src/components/component.css [hunk 2]
```diff
@@ -138,6 +138,11 @@ a.button:visited {
   margin-left: 2vh;
 }
 
+/* :TODO figure out what is overriding */
+.deleteIcon {
+  width: 3vh !important;
```
Comment (aseehra) [inline src/components/component.css]: ew, `!important`.

Comment (ender554) [reply src/components/component.css]: AGREED!

File: src/components/ShoppingList/ShoppingList.js (hunk 1 above)
Comment (ender554) [reply src/components/ShoppingList/ShoppingList.js]: thanks!

File: src/components/ShoppingList/ShoppingList.js [hunk 3]
```diff
@@ -1,28 +1,78 @@
...
+import React, { Component } from 'react';
 import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { deleteList } from '../../actions/shoppingLists';
 import '../component.css';
 
-export default function ShoppingList({ id, name, store, editing }) {
-  if (editing) {
+export class ShoppingList extends Component {
+  // export default function ShoppingList({ id, name, store, editing }) {
```
[the remark this answers is not in the document]
Comment (ender554) [reply src/components/ShoppingList/ShoppingList.js]: done

File: src/actions/shoppingLists.js [hunk 4]
```diff
@@ -17,6 +17,49 @@ export const listsError = error => ({
...
+export const deleteList = listId => (dispatch, getState) => {
+  dispatch(deleteListRequest());
+  const authToken = getState().auth.authToken;
+  return fetch(`${API_BASE_URL}/api/lists/${listId}`, {
+    method: 'DELETE',
+    headers: {
+      Authorization: `Bearer ${authToken}`,
+    },
+  })
+    .then(res => {
```
[the remark this answers is not in the document]
Comment (ender554) [reply src/actions/shoppingLists.js]: switched!

**Pull request merged**